modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/events/LoginPostAction.java

```diff
@@ -111,30 +157,111 @@ private void _addDefaultAccountRoles(HttpServletRequest httpServletRequest)
 		}
 	}
 
-	private AccountEntry _getAccountEntry(
-		int commerceSiteType, HttpServletRequest httpServletRequest,
-		long userId) {
+	private void _associateAccountToOrder(
+			AccountEntry accountEntry, long commerceChannelGroupId,
+			CommerceOrder commerceOrder, HttpServletRequest httpServletRequest,
+			long userId)
+		throws PortalException {
+
+		CommerceOrder userCommerceOrder =
+			_commerceOrderLocalService.fetchCommerceOrder(
+				accountEntry.getAccountEntryId(), commerceChannelGroupId,
+				userId, CommerceOrderConstants.ORDER_STATUS_OPEN);
+
+		if (userCommerceOrder != null) {
+			CommerceContext commerceContext = _commerceContextFactory.create(
+				_portal.getCompanyId(httpServletRequest),
+				commerceChannelGroupId, userId,
+				userCommerceOrder.getCommerceOrderId(),
+				accountEntry.getAccountEntryId());
+
+			ServiceContext serviceContext = ServiceContextFactory.getInstance(
+				httpServletRequest);
+
+			PermissionThreadLocal.setPermissionChecker(
+				PermissionCheckerFactoryUtil.create(
+					_portal.getUser(httpServletRequest)));
+
+			_commerceOrderLocalService.mergeGuestCommerceOrder(
+				userId, commerceOrder.getCommerceOrderId(),
+				userCommerceOrder.getCommerceOrderId(), commerceContext,
+				serviceContext);
+		}
+		else {
+			_commerceOrderLocalService.updateAccount(
+				commerceOrder.getCommerceOrderId(), userId,
+				accountEntry.getAccountEntryId());
+		}
+	}
+
+	private AccountEntry _createAccountEntry(
+			long commerceChannelGroupId, String name, String type, User user)
+		throws PortalException {
 
+		ServiceContext serviceContext = new ServiceContext();
+
+		long userId = user.getUserId();
+
+		serviceContext.setCompanyId(user.getCompanyId());
+		serviceContext.setUserId(userId);
+
+		AccountEntry accountEntry = _accountEntryLocalService.addAccountEntry(
+			userId, AccountConstants.PARENT_ACCOUNT_ENTRY_ID_DEFAULT, name,
+			null, null, user.getEmailAddress(), null, StringPool.BLANK, type,
+			WorkflowConstants.STATUS_APPROVED, serviceContext);
+
+		_accountEntryUserRelLocalService.addAccountEntryUserRel(
+			accountEntry.getAccountEntryId(), userId);
+
+		if (type.equals(AccountConstants.ACCOUNT_ENTRY_TYPE_BUSINESS)) {
+			_addBusinessAccountRoles(accountEntry, user);
+		}
+
+		int count =
+			_commerceChannelAccountEntryRelLocalService.
+				getCommerceChannelAccountEntryRelsCount(
+					commerceChannelGroupId, null,
+					CommerceChannelAccountEntryRelConstants.TYPE_ELIGIBILITY);
+
+		if (count > 0) {
+			_commerceChannelAccountEntryRelLocalService.
+				addCommerceChannelAccountEntryRel(
+					userId, accountEntry.getAccountEntryId(), null, 0,
+					commerceChannelGroupId, false, 0,
+					CommerceChannelAccountEntryRelConstants.TYPE_ELIGIBILITY);
+		}
```

Comment:
remove as unneeded, if eligibility is in place guests would be automatically unable to access the channel